src/test/app/Batch_test.cpp

```diff
@@ -510,207 +533,288 @@ class Batch_test : public beast::unit_test::suite
+            env(pay(gw, bob, USD(100)));
+            env.close();
+
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+
+            auto const seq = env.seq(alice);
+            auto const ammCreate =
+                [&alice](STAmount const& amount, STAmount const& amount2) {
+                    Json::Value jv;
+                    jv[jss::Account] = alice.human();
+                    jv[jss::Amount] = amount.getJson(JsonOptions::none);
+                    jv[jss::Amount2] = amount2.getJson(JsonOptions::none);
+                    jv[jss::TradingFee] = 0;
+                    jv[jss::TransactionType] = jss::AMMCreate;
+                    return jv;
+                };
+
+            auto const batchFee = calcBatchFee(env, 0, 2);
+            env(batch::batch(alice, seq, batchFee, tfAllOrNothing),
+                batch::add(ammCreate(XRP(10), USD(10)), seq + 1),
+                batch::add(pay(alice, bob, XRP(10)), seq + 2),
+                ter(telINSUF_FEE_P));
+            env.close();
+
+            BEAST_EXPECT(env.seq(alice) == seq);
+            BEAST_EXPECT(env.balance(alice) == preAlice);
+            BEAST_EXPECT(env.balance(bob) == preBob);
+        }
     }
 
     void
-    testBadFeeSigner(FeatureBitset features)
+    testChangesBetweenViews(FeatureBitset features)
     {
-        testcase("bad fee signer");
+        testcase("changes between views");
 
         using namespace test::jtx;
         using namespace std::literals;
 
         test::jtx::Env env{*this, envconfig()};
 
-        auto const feeDrops = env.current()->fees().base;
         auto const alice = Account("alice");
         auto const bob = Account("bob");
-        auto const gw = Account("gw");
-        auto const USD = gw["USD"];
-
-        env.fund(XRP(1000), alice, bob, gw);
-        env.close();
-        env.trust(USD(1000), alice, bob);
-        env(pay(gw, alice, USD(100)));
-        env(pay(gw, bob, USD(100)));
-        env.close();
-
-        env(noop(bob), ter(tesSUCCESS));
+        auto const carol = Account("carol");
+        env.fund(XRP(220), alice, bob, carol);
         env.close();
 
-        auto const preAliceSeq = env.seq(alice);
         auto const preAlice = env.balance(alice);
-        auto const preAliceUSD = env.balance(alice, USD.issue());
-        auto const preBobSeq = env.seq(bob);
         auto const preBob = env.balance(bob);
-        auto const preBobUSD = env.balance(bob, USD.issue());
 
-        auto const batchFee = ((0 + 2) * feeDrops) + feeDrops * 2;
-        env(batch::batch(alice, preAliceSeq, batchFee, tfAllOrNothing),
-            batch::add(pay(alice, bob, XRP(10)), alice, 1, preAliceSeq),
-            batch::add(pay(bob, alice, XRP(5)), alice, 0, preBobSeq),
-            batch::sig(bob),
-            ter(telINSUF_FEE_P));
+        // Using 1 XRP to create insufficient reserve result
+        auto const batchFee = XRP(1);
```

Comment:
ditto. we should specify the `batchFee`  relative to the fee variables, so it's easier to understand, something like

```
auto const acctReserve = env.current()->fees().accountReserve(0);
auto const incReserve = env.current()->fees().increment;
auto const fee = env.current()->fees().base;
	...
	...
	...
auto const batchFee = fee * 2; // Just making this up, should change the value accordingly
```

Reply:
fixed